src/components/calcite-slider/calcite-slider.tsx

```diff
@@ -102,98 +548,63 @@ export class CalciteSlider {
-        {this.isRange ? (
-          <button
-            ref={(el) => (this.minHandle = el as HTMLButtonElement)}
-            onFocus={() => (this.activeProp = "minValue")}
-            onBlur={() => (this.activeProp = null)}
-            onMouseDown={() => this.dragStart("minValue")}
-            onTouchStart={(e) => this.dragStart("minValue", e)}
-            role="slider"
-            aria-orientation="horizontal"
-            aria-label={this.minLabel}
-            aria-valuenow={this.minValue}
-            aria-valuemin={this.min}
-            aria-valuemax={this.max}
-            disabled={this.disabled}
-            style={{ left }}
-            class={{
-              thumb: true,
-              "thumb--min": true,
-              "thumb--active": this.dragProp === "minValue",
-              "thumb--precise": this.precise,
-            }}
-          >
-            <span class="handle"></span>
-            {this.labelHandles ? (
-              <span class="handle__label" aria-hidden="true">
-                {this.minValue}
-              </span>
-            ) : (
-              ""
-            )}
-          </button>
-        ) : (
-          ""
-        )}
-        <button
-          ref={(el) => (this.maxHandle = el as HTMLButtonElement)}
-          onFocus={() => (this.activeProp = maxProp)}
-          onBlur={() => (this.activeProp = null)}
-          onMouseDown={() => this.dragStart(maxProp)}
-          onTouchStart={(e) => this.dragStart(maxProp, e)}
-          role="slider"
-          aria-orientation="horizontal"
-          aria-label={this.isRange ? this.maxLabel : this.minLabel}
-          aria-valuenow={this[maxProp]}
-          aria-valuemin={this.min}
-          aria-valuemax={this.max}
-          disabled={this.disabled}
-          style={{ right }}
-          class={{
-            thumb: true,
-            "thumb--max": true,
-            "thumb--active":
-              this.lastDragProp !== "minMaxValue" && this.dragProp === maxProp,
-            "thumb--precise": this.precise,
-          }}
-        >
-          <span class="handle"></span>
-          {this.labelHandles ? (
-            <span class="handle__label" aria-hidden="true">
-              {this[maxProp]}
-            </span>
-          ) : (
-            ""
-          )}
-        </button>
+        {!this.precise && !this.labelHandles && this.isRange && minHandle}
+        {!this.hasHistogram &&
+          !this.precise &&
+          this.labelHandles &&
+          this.isRange &&
+          minLabeledHandle}
+        {this.precise && !this.labelHandles && this.isRange && minPreciseHandle}
+        {this.precise &&
+          this.labelHandles &&
+          this.isRange &&
+          minLabeledPreciseHandle}
+        {this.hasHistogram &&
+          !this.precise &&
+          this.labelHandles &&
+          this.isRange &&
+          minHistogramLabeledHandle}
+
+        {!this.precise && !this.labelHandles && handle}
+        {!this.hasHistogram &&
```

Comment:
can this be extracted into a `getHandle` function or something? This is really hard to read, IMO

Reply:
I agree, It is hard to read and quite frankly a mess, but I found that it would be more jumping around in the code to see what's really going on to pull them out into functions and during development of this it was handy to be able to see all the cases in one chunk of code because they all kinda inform the others in a way based on all the possible prop combinations.